internal/pkg/prometheus/prometheus.go

```diff
@@ -62,16 +49,17 @@ func Get(client HTTPDoer, url string) (MetricFamiliesByName, error) {
 		return mfs, err
 	}
 
-	defer func() {
-		_ = resp.Body.Close()
-	}()
-
 	if resp.StatusCode < 200 || resp.StatusCode > 300 {
 		return nil, fmt.Errorf("status code returned by the prometheus exporter indicates an error occurred: %d", resp.StatusCode)
 	}
 
-	countedBody := &countReadCloser{innerReadCloser: resp.Body}
-	d := expfmt.NewDecoder(countedBody, expfmt.FmtText)
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return mfs, err
+	}
+	b := bytes.NewBuffer(body)
```

Comment:
Perhaps we could use `bytes.NewReader` rather than `NewBuffer`, which afaik it's cheaper :)

Reply:
ooh I'll give that a shot! 💡